core/toolbox.js

```diff
@@ -315,18 +332,39 @@ Blockly.Toolbox.prototype.getSelectedItem = function() {
  */
 Blockly.Toolbox.prototype.setSelectedItem = function(item) {
   if (this.selectedItem_) {
-    // Don't do anything if they selected the already-open category.
-    if (this.selectedItem_ == item) {
-      return;
-    }
     // They selected a different category but one was already open.  Close it.
     this.selectedItem_.setSelected(false);
   }
   this.selectedItem_ = item;
   if (this.selectedItem_ != null) {
     this.selectedItem_.setSelected(true);
-    this.flyout_.show(item.getContents());
-    this.flyout_.scrollToStart();
+    // Scroll flyout to the top of the selected category
+    var categoryName = item.name_;
+    for (var i=0; i<this.flyout_.buttons_.length; i++) {
+      if (this.flyout_.buttons_[i].text_ === categoryName) {
+        if (this.horizontalLayout_) {
+          this.flyout_.scrollTo(this.flyout_.buttons_[i].position_.x);
```

Comment:
You're looking at a private member (position_) of a private member (buttons_), and you're doing it all over the place in this chunk of code.  Please figure out what information you want made accessible where, and then try to get it without just reaching through arbitrary private pieces of code.  If you need a position and button stored for every category, maybe you should store those in the Blockly.Toolbox.Category instances and keep them updated.

Reply:
This is fixed